eliot-service/eliot/symbolicate_resource.py

```diff
@@ -392,114 +376,148 @@ def symbolicate(self, jobs, debug_stats):
 
         """
 
-        results = []
-        for i, job in enumerate(jobs):
-            stacks = job.get("stacks", [])
-            modules = job.get("memoryMap", [])
-
-            # Build list of Module instances so we can keep track of what we've
-            # used/seen
-            module_records = [
-                ModuleInfo(
-                    filename=debug_filename,
-                    debug_filename=debug_filename,
-                    debug_id=debug_id,
-                    has_symcache=None,
-                    symcache=None,
-                )
-                for debug_filename, debug_id in modules
-            ]
-
-            symbolicated_stacks = []
-            for stack_index, stack in enumerate(stacks):
-                METRICS.histogram("eliot.symbolicate.frames_count", value=len(stack))
-                symbolicated_stack = []
-                for frame_index, frame in enumerate(stack):
+        # List of:
+        #
+        # ((debug_filename, debug_id), frame_info)
+        #
+        # tuples
+        frames = []
+
+        job_results = []
+
+        # Go through jobs, stacks, and frames and build a list of all the things to
+        # symbolicate
+        for job_i, job in enumerate(jobs):
+            # List of stacks
+            stacks = job["stacks"]
+
+            # List of [debug_filename, debug_id] lists
+            memorymap = job["memoryMap"]
+
+            stacks_results = []
+            for stack_i, stack in enumerate(stacks):
+                stack_results = []
+                for frame_i, frame in enumerate(stack):
                     module_index, module_offset = frame
-                    module_info = None
-                    data = {
-                        "frame": frame_index,
-                        "module": "<unknown>",
+
+                    if 0 <= module_index < len(memorymap):
+                        debug_filename, debug_id = memorymap[module_index]
+                    else:
+                        debug_filename, debug_id = None, None
```

Comment:
I should switch this to use empty strings because those are falsey and the type of these variables is consistent.